File: expectanim/src/main/java/com/github/florent37/expectanim/core/position/PositionAnimExpectationTopOfParent.java
package com.github.florent37.expectanim.core.position;

import android.view.View;

/**
 * Created by florentchampigny on 17/02/2017.
 */

public class PositionAnimExpectationTopOfParent extends PositionAnimExpectation {

    public PositionAnimExpectationTopOfParent() {
        setForPositionY(true);
    }

    @Override
    public Float getCalculatedValueX(View viewToMove) {
        return null;
    }

    @Override
    public Float getCalculatedValueY(View viewToMove) {
        return getMargin(viewToMove);
    }
}

File: kotlinpleaseanimate/src/main/java/com/github/florent37/kotlin/pleaseanimate/core/position/PositionAnimExpectationTopOfParent.kt
package com.github.florent37.kotlin.pleaseanimate.core.position

import android.view.View

class PositionAnimExpectationTopOfParent : PositionAnimExpectation() {

    init {
        isForPositionY = true
    }

    override fun getCalculatedValueX(viewToMove: View): Float? {
        return null
    }

    override fun getCalculatedValueY(viewToMove: View): Float? {
        return getMargin(viewToMove)
    }
}
